modules/auxiliary/scanner/http/suite_crm_export_sqli.rb

```diff
@@ -0,0 +1,244 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+
+  include Msf::Auxiliary::Scanner
+  include Msf::Exploit::SQLi
+  include Msf::Exploit::Remote::HttpClient
+
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'SuiteCRM authenticated SQL injection in export functionality',
+        'Description' => %q{
+          This module exploits an authenticated SQL injection in SuiteCRM in versions before 7.12.6. The vulnerability
+          allows an attacker to send specically crafted requests to the export entry point of the application in order
+          to retrieve all the usernames and their associated password from the database.
+        },
+        'Author' => [
+          'Exodus Intelligence', # Advisory
+          'jheysel-r7', # poc + msf module
+          'Redouane NIBOUCHA <rniboucha@yahoo.fr>' # sql injection help
+        ],
+        'License' => MSF_LICENSE,
+        'References' => [
+          ['URL', 'https://blog.exodusintel.com/2022/06/09/salesagility-suitecrm-export-request-sql-injection-vulnerability/']
+        ],
+        'Actions' => [
+          ['Dump tables', { 'Description' => 'Dumps database tables' }]
+        ],
+        'DefaultAction' => 'Dump tables',
+        'DisclosureDate' => '2022-06-09',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'SideEffects' => [IOC_IN_LOGS],
+          'Reliability' => [REPEATABLE_SESSION]
+        },
+        'Privileged' => true
+      )
+    )
+    register_options [
+      OptInt.new('COUNT', [false, 'Number of users to enumerate', 3]),
+      OptString.new('USERNAME', [true, 'Username of user', 'normal_user']),
+      OptString.new('PASS', [true, 'Password for user', 'normal_user']),
+    ]
+  end
+
+  def check
+    authenticated = authenticate
+    fail_with Failure::NoAccess, 'Unable to authenticate to SuiteCRM' unless authenticated
```

Comment:
You cannot use `fail_with` inside of a `check` method or any of the methods that the `check` method calls. All `check` methods must return a `Exploit::CheckCode` value only. Please reorganize your code to accommodate this requirement.